File: chart_scheduler/db/models/kr_ticker_charts_model.py
from datetime import datetime
from sqlalchemy import BigInteger, Column, Date, DateTime, ForeignKey, Index, Integer, String , Float , DECIMAL
import sqlalchemy
from sqlalchemy.orm import relationship , backref
from chart_scheduler.db.session import Base , meta_obj
from chart_scheduler.constants.schema_names import SCHEMA_NAMES

class KrTickerChartModel(Base) :
  __tablename__ = SCHEMA_NAMES.KR_TICKER_CHARTS
  RELATIONSHIPS_TO_DICT = True
  __metaclass__=meta_obj
  __table_args__=(Index('partition_idx' , 'symbol' , 'date'),)

  symbol = Column(
    String(12), 
    index=True,
    primary_key=True
  )
  ticker_symbol = Column(
    String(12),
    nullable=False
  )
  date : datetime = Column(
    Date , 
    nullable=False , 
    primary_key=True,
    index=True,
  )
  open : int = Column(DECIMAL(precision=11 , scale=3) , nullable=False)
  high : int = Column(DECIMAL(precision=11 , scale=3) , nullable=False)
  low : int = Column(DECIMAL(precision=11 , scale=3) , nullable=False)
  close : int = Column(DECIMAL(precision=11 , scale=3) , nullable=False)
  volume : int = Column(BigInteger , nullable=False)
  percent : float = Column(Float , nullable=False , default=0)
  ticker = relationship(
    'KrTickerModel', # class name
    back_populates='charts',
  )

File: gin/common/models/kr_ticker_charts_model.go
package models

import (
	"time"

	"github.com/jjh930301/market/common/constants"
	"github.com/shopspring/decimal"
)

type KrTickerChartsModel struct {
	Symbol       string `gorm:"column:symbol;type:varchar(12);primary_key;index" json:"symbol,omitempty"` //index
	TickerSymbol string `gorm:"column:ticker_symbol;type:varchar(12);not null" json:"ticker_symbol,omitempty"`
	Ticker       KrTickerModel
	Date         time.Time       `gorm:"column:date;type:date;primary_key;index" json:"date,omitempty"`
	Open         decimal.Decimal `gorm:"column:open;type:decimal(11,3)" json:"open,omitempty"`
	High         decimal.Decimal `gorm:"column:high;type:decimal(11,3)" json:"high,omitempty"`
	Low          decimal.Decimal `gorm:"column:low;type:decimal(11,3)" json:"low,omitempty"`
	Close        decimal.Decimal `gorm:"column:close;type:decimal(11,3)" json:"close,omitempty"`
	Percent      float32         `gorm:"column:percent;type:float;not null" json:"percent,omitempty"`
	Volume       int64           `gorm:"column:volume;type:bigint;not null" json:"volume,omitempty"`
}

func (KrTickerChartsModel) TableName() string {
	return constants.KrTickerChartTableName
}
